### ISSUE TITLE
API consistency: "val" vs "validation"

### ISSUE THREAD

**Issue by msegado** [PyTorchLightning/pytorch-lightning#6094]
### Motivation

As a new user learning PyTorch Lightning, I was surprised at the naming inconsistency between the `validation_step()` and `val_dataloader()` hooks. Searching through the codebase, it seems that *most* methods or getters/setters favor "validation"...

- `validation_step_end`
- `validation_step`
- `validation_epoch_end`
- `on_validation_start`
- `on_validation_model_train`
- `on_validation_model_eval`
- `on_validation_epoch_start`
- `on_validation_epoch_end`
- `on_validation_end`
- `on_validation_batch_start`
- `on_validation_batch_end`
- `init_validation_tqdm`
- `enable_validation`
- `disable_validation`
- `_on_validation_start_log`
- `_on_validation_epoch_start_log`
- `_on_validation_epoch_end_log`
- `_on_validation_end_log`
- `_on_validation_batch_start_log`
- `_on_validation_batch_end_log`

...while a handful use "val" instead:
- `val_transforms`
- `val_step_context`
- `val_dataloader`
- `val_batch_idx`
- `total_val_batches`
- `should_check_val_fx`
- `reset_val_dataloader`
- `reset_train_val_dataloaders`
- `num_seen_val_check_batches`

I wasn't able to see any obvious pattern in which methods got which naming convention, and think having a consistent convention throughout would make for a (slightly) cleaner and easier-to-learn library.

Related: https://github.com/PyTorchLightning/pytorch-lightning/pull/634

### Pitch

Standardize all validation-related methods (and properties, if any exist) to use the same naming convention; either "val" (for consistency with e.g. torchvision) or "validation" (for consistency with the bulk of the current Lightning API). Renamed methods would be marked as deprecated and removed in the next major version bump.

**Comment by Borda** [comment]
cc: @PyTorchLightning/core-contributors ^^

**Comment by s-rog** [comment]
If we were to implement this change, I would prefer `val` for its brevity.

**Comment by justusschock** [comment]
We have the same in other parts as well... train vs. training, test vs testing

**Comment by carmocca** [comment]
Poll:

**Comment by msegado** [comment]
@justusschock True (and also predict vs. predicting vs. prediction). These are a bit more confusing since the short forms are also valid imperative verbs, though looking at the codebase they do seem to be used mostly as abbreviations, at least in the case of "train" vs. "training"; e.g. `train_dataloader` refers to "the dataloader to use during _training_," not to a command to "_train_ the dataloader".

@Borda, do you think it makes sense to expand the scope of this issue to include the variants of train*/test*/predict* as well? Also, I'm all for API consistency but don't want this to devolve into bikeshedding about method names, so if there are more appropriate forums than GitHub for tackling this please let us know 🙂.

PS: For reference, here's a (likely incomplete) list of train*, test*, and predict* methods/getters/setters: 

<details>
<summary>
[click to expand]
</summary>

"training":
- `training`
- `training_type_plugin`
- `training_step`
- `training_step_end`
- `training_step_and_backward`
- `training_epoch_end`
- `start_training`
- `should_skip_training`
- `select_training_type_plugin`
- `run_training_epoch`
- `run_training_batch`
- `restore_training_state`
- `resolve_training_type_plugin`
- `post_training_step`
- `freeze_before_training`
- `connect_training_type_plugin`
- `cache_training_step_metrics`
- `_process_training_step_output`
- `_process_training_step_output_1_0`
- `_pre_training_routine`
- `_num_training_batches_reached`
- `_check_training_step_output`
- `__save_end_of_training_weights`
- `__run_legacy_training_epoch_end`

"train":
- `update_train_loop_lr_schedulers`
- `train_transforms`
- `train_step_context`
- `train_step_and_backward_closure`
- `train_or_test_or_predict`
- `train_dataloader`
- `train_batch_idx`
- `track_train_loss_history`
- `total_train_batches`
- `save_loggers_on_train_batch_end`
- `run_train`
- `run_train_split_start`
- `reset_train_val_dataloaders`
- `reset_train_dataloader`
- `process_train_step_outputs`
- `on_validation_model_train`
- `on_train_start`
- `on_train_split_start`
- `on_train_epoch_start`
- `on_train_epoch_end`
- `on_train_end`
- `on_train_batch_start`
- `on_train_batch_end`
- `on_test_model_train`
- `on_evaluation_model_train`
- `log_train_step_metrics`
- `log_train_epoch_end_metrics`
- `init_train_tqdm`
- `get_profiled_train_dataloader`
- `build_train_args`
- `_on_train_start_log`
- `_on_train_epoch_start_log`
- `_on_train_epoch_end_log`
- `_on_train_end_log`
- `_on_train_batch_start_log`
- `_on_train_batch_end_log`
- `_initialize_deepspeed_train`
- `__verify_train_loop_configuration`

"testing":
- `testing`
- `start_testing`

"test":
- `train_or_test_or_predict`
- `total_test_batches`
- `test`
- `test_transforms`
- `test_step`
- `test_step_end`
- `test_step_context`
- `test_epoch_end`
- `test_dataloader`
- `test_batch_idx`
- `run_test`
- `reset_test_dataloader`
- `on_test_start`
- `on_test_model_train`
- `on_test_model_eval`
- `on_test_epoch_start`
- `on_test_epoch_end`
- `on_test_end`
- `on_test_batch_start`
- `on_test_batch_end`
- `num_seen_test_check_batches`
- `init_test_tqdm`
- `has_setup_test`
- `_on_test_start_log`
- `_on_test_epoch_start_log`
- `_on_test_epoch_end_log`
- `_on_test_end_log`
- `_on_test_batch_start_log`
- `_on_test_batch_end_log`
- `__test_using_best_weights`
- `__test_given_model`

"predicting":
- `start_predicting`
- `predicting`

"prediction":
- `write_prediction_dict`
- `write_prediction`
- `_add_prediction`

"predict":
- `train_or_test_or_predict`
- `total_predict_batches`
- `should_skip_predict`
- `run_predict`
- `reset_predict_dataloader`
- `predict_dataloader`
- `predict_context`
- `predict_batch_idx`
- `predict`
- `on_predict_start`
- `on_predict_model_eval`
- `on_predict_epoch_end`
- `on_predict_end`
- `on_predict_batch_end`
- `init_predict_tqdm`
- `get_predict_dataloaders`
</details>

**Comment by Borda** [comment]
just feel that using `val` may collide with a shortcut for value... :confused:

**Comment by carmocca** [comment]
> just feel that using `val` may collide with a shortcut for value... 😕

All cases are `val_something`, not just `val`

**Comment by amrutha1098** [comment]
Hi, Is this issue still up for grab ?

**Comment by justusschock** [comment]
I am not sure, whether we currently want to change this. The  APIs are quite stable right now and people got used to these minor inconsistencies.

Thoughts @carmocca @awaelchli

**Comment by carmocca** [comment]
I agree. Let's try to be consistent with any future additions, but be happy with these inconsistencies done in the past.

**Comment by Borda** [comment]
cc: @lantiga for your thoughts?
